packages/blockly-field-slider/src/field_slider.js

```diff
@@ -0,0 +1,196 @@
+ * @constructor
+ */
+export const FieldSlider = function(opt_value, opt_min, opt_max, opt_precision,
+    opt_validator, opt_config) {
+  FieldSlider.superClass_.constructor.call(this, opt_value, opt_min,
+      opt_max, opt_precision, opt_validator, opt_config);
+};
+Blockly.utils.object.inherits(FieldSlider, Blockly.FieldNumber);
+
+/**
+ * Constructs a FieldSlider from a JSON arg object.
+ * @param {!Object} options A JSON object with options (value, min, max, and
+ *                          precision).
+ * @return {!FieldSlider} The new field instance.
+ * @package
+ * @nocollapse
+ */
+FieldSlider.fromJson = function(options) {
+  return new FieldSlider(options['value'],
+      undefined, undefined, undefined, undefined, options);
+};
+
+/**
+ * Creates and shows the slider field's editor.
+ * @protected
+ * @override
+ */
+FieldSlider.prototype.showEditor_ = function() {
+  // Build the DOM.
+  var editor = this.dropdownCreate_();
+
+  Blockly.DropDownDiv.getContentDiv().appendChild(editor);
+
+  Blockly.DropDownDiv.setColour(this.sourceBlock_.style.colourPrimary,
+      this.sourceBlock_.style.colourTertiary);
+
+  Blockly.DropDownDiv.showPositionedByField(
+      this, this.dropdownDispose_.bind(this));
+};
+
+/**
+ * Creates the slider editor and add event listeners.
+ * @return {!Element} The newly created slider.
+ * @private
+ */
+FieldSlider.prototype.dropdownCreate_ = function() {
+  var wrapper = document.createElement('div');
+  var sliderInput = document.createElement('input');
+  sliderInput.setAttribute('type', 'range');
+  sliderInput.setAttribute('min', this.min_);
+  sliderInput.setAttribute('max', this.max_);
+  sliderInput.setAttribute('value', this.getValue());
+  sliderInput.className ='fieldSlider';
+  wrapper.appendChild(sliderInput);
+  this.sliderInput_ = sliderInput;
+
+  this.sliderListener_ = Blockly.bindEventWithChecks_(
+      sliderInput, 'input', this, this.onSliderChange_);
+  return wrapper;
+};
+
+/**
+ * Disposes of events belonging to the slider editor.
+ * @private
+ */
+FieldSlider.prototype.dropdownDispose_ = function() {
+  this.sliderInput_ = null;
+  Blockly.unbindEvent_(this.sliderListener_);
+};
+
+/**
+ * Updates the slider when the field rerenders.
+ * @private
+ * @override
+ */
+FieldSlider.prototype.render_ = function() {
+  FieldSlider.superClass_.render_.call(this);
+  this.updateSlider_();
+};
+
+/**
+ * Sets the text to match the slider's position.
+ * @param {!Event} e Mouse move event.
+ * @protected
+ */
+FieldSlider.prototype.onSliderChange_ = function(e) {
+  this.setEditorValue_(this.sliderInput_.value);
+};
+
+/**
+ * Updates the slider when the field rerenders.
+ * @private
+ * @override
+ */
+FieldSlider.prototype.updateSlider_ = function() {
+  if (!this.sliderInput_) {
+    return;
+  }
+  this.sliderInput_.setAttribute('value', this.getValue());
+};
+
+Blockly.fieldRegistry.register('field_slider', FieldSlider);
+
+/**
+ * CSS for slider field.
+ */
+Blockly.Css.register([
```

Comment:
Since we're transpiling, do we also get multiline strings?

Reply:
Sure do.

Reply:
Done